.github/actions/make-version-changes/script.js

```diff
@@ -0,0 +1,223 @@
+const fs = require('fs');
+const path = require('path');
+const { execSync } = require('child_process');
+
+// todo: move somewhere, like a separate config/constants file.
+const MPL_PROGRAM_CONFIG = {
+  'auction-house': {
+    has_idl: true,
+    uses_anchor: true,
+  },
+  auction: {
+    has_idl: false,
+    uses_anchor: false,
+  },
+  auctioneer: {
+    has_idl: true,
+    uses_anchor: true,
+  },
+  core: {
+    has_idl: false,
+    uses_anchor: false,
+  },
+  'candy-machine': {
+    has_idl: true,
+    uses_anchor: true,
+  },
+  'fixed-price-sale': {
+    has_idl: true,
+    uses_anchor: true,
+  },
+  gumdrop: {
+    has_idl: false,
+    uses_anchor: false,
+  },
+  metaplex: {
+    has_idl: false,
+    uses_anchor: false,
+  },
+  'nft-packs': {
+    has_idl: false,
+    uses_anchor: false,
+  },
+  'token-entangler': {
+    has_idl: true,
+    uses_anchor: true,
+  },
+  // uses shank
+  'token-metadata': {
+    has_idl: true,
+    uses_anchor: false,
+  },
+  // uses shank
+  'token-vault': {
+    has_idl: true,
+    uses_anchor: false,
+  },
+};
+
+const wrappedExec = (cmd, cwd) => {
+  let args = {
+    stdio: 'inherit',
+  };
+
+  if (cwd) {
+    args['cwd'] = cwd;
+  } else {
+    // default to curernt dir
+    args['cwd'] = path.join(__dirname);
+  }
+
+  execSync(cmd, args);
+};
+
+const packageUsesAnchor = (pkg) => {
+  const result = MPL_PROGRAM_CONFIG[pkg]['uses_anchor'];
+  console.log(`${pkg} uses anchor: ${result}`);
+  return result;
+};
+
+const packageHasIdl = (pkg) => {
+  const result = MPL_PROGRAM_CONFIG[pkg]['has_idl'];
+  console.log(`${pkg} has idl: ${result}`);
+  return result;
+};
+
+const isPackageType = (actual, target) => actual === target;
+// additional equality checks can match other subdirs, e.g. `rust|test|cli|<etc>`
+const isCratesPackage = (actual) => isPackageType(actual, 'program');
+const isNpmPackage = (actual) => isPackageType(actual, 'js');
+
+const parseVersioningCommand = (cmd) => cmd.split(':');
+const shouldUpdate = (actual, target) => target === '*' || target === actual;
+
+const updateCratesPackage = async (io, cwdArgs, pkg, semvar) => {
+  console.log('updating rust package');
+  const currentDir = cwdArgs.join('/');
+
+  // adds git info automatically
+  wrappedExec(
+    `cargo release --no-publish --no-push --no-confirm --verbose --execute --no-verify --no-tag --config ../../release.toml ${semvar}`,
+    currentDir,
+  );
+
+  const sourceIdlDir = [...cwdArgs.slice(0, cwdArgs.length - 2), 'target', 'idl'].join('/');
+
+  // generate IDL
+  if (packageHasIdl(pkg)) {
```

Comment:
We synced on this as well - the purpose of this is purely to update the version in the IDL, not generate a new IDL. So, maybe we can save **a lot** of time by purely editing the IDL JSON instead of installing both anchor and shank-cli.

Reply:
But, this reminded me about @thlorenz idea for a workflow to check the output of `yarn api:gen` when a PR is opened. I added a task in linear for this.